### DISCUSSION TITLE
Warning about 'Failed to detect analyzer binary!'

### DISCUSSION THREAD

**Discussion by nicsav** [Ericsson/codechecker#3584] [Q&A] [answered]
Hi all, I ran:

CodeChecker analyze compile_commands.json -o ./reports

and got something like this:

[snip]
[INFO 2022-01-21 10:36] - ----==== Summary ====----
[INFO 2022-01-21 10:36] - Successfully analyzed
[INFO 2022-01-21 10:36] -   clang-tidy: 167
[INFO 2022-01-21 10:36] - Total analyzed compilation commands: 167
[INFO 2022-01-21 10:36] - ----=================----
[INFO 2022-01-21 10:36] - Analysis finished.
[INFO 2022-01-21 10:36] - To view results in the terminal use the "CodeChecker parse" command.
[INFO 2022-01-21 10:36] - To store results use the "CodeChecker store" command.
[INFO 2022-01-21 10:36] - See --help and the user guide for further options about parsing and storing the reports.
[INFO 2022-01-21 10:36] - ----=================----
[INFO 2022-01-21 10:36] - Analysis length: 439.2375319004059 sec.
[WARNING 2022-01-21 10:36] - Analyzer 'clangsa' is enabled but CodeChecker is failed to execute analysis with it: 'Failed to detect analyzer binary!'. Please check your 'PATH' environment variable and the 'config/package_layout.json' file!

I was able to run

CodeChecker parse ./reports -e html -o ./reports_html

after that and it seemed to generate a bunch of reports.

However, I'm still not sure what the warning was about and if it affected the amount and quality of reports. The message itself is far from being helpful in debugging this as it not clear on what exactly failed and does not say exactly what to look for.

Any thoughts?

Nick

**Comment by csordasmarton** [comment, marked as answer]
`CodeChecker analyze` command by default will try to run both [clang-tidy](http://clang.llvm.org/extra/clang-tidy/) and [clangsa](http://clang-analyzer.llvm.org/) analyzers. In your system it looks like you have installed only `clang-tidy` so CodeChecker will only run only this analyzer. 

If you don't want to see this warning message you can enable only `clang-tidy` analyzer (`CodeChecker analyze --analyzers clang-tidy ...`) or you can install clangsa analyzer (for more information about this see: https://clang-analyzer.llvm.org/).

**Comment by nicsav** [comment]
Ah, indeed. I built it clangsa from scratch and the warning has gone away.

**Comment by ralwing** [comment]
Just for future references: According to mentioned above [instruction](https://clang-analyzer.llvm.org/installation.html), it is required to build the clang from source, which is in fact not necessarily truth. In ubuntu CodeChecker launches `clang --analyze`, so in fact the only required to install clang package:
`apt install clang`

**Reply by whisperity** [reply]
http://clang-analyzer.llvm.org is very, **very** outdated. Like 6 years of outdated. It is not even part of the new RST/Sphinx based official LLVM documentation system. Many of the actively worked on documentation (like the checkers themselves and CTU) is at http://clang.llvm.org/docs/.

If you are using a Debian or Ubuntu-based system and can afford to change system-level details, you might even want to use the nightly PPA: http://apt.llvm.org/. Stable release builds, newer than what is available in the official sources most of the time, can also be grabbed from here!
